File: examples/extending-objects/Inheritance/Python/employee.py
# Example retrieved from https://www.python-course.eu/python3_inheritance.php
# Date retrieved: Nov 15, 2018
# Example slightly modified

from person import Person
class Employee(Person):

    def __init__(self, first, last, age, staffnum):
        super().__init__(first, last, age)
        self.staffnumber = staffnum

    def __str__(self):
        return super().__str__() + ", " +  self.staffnumber

File: examples/extending-objects/Inheritance/Go/employee.go
package main

import "fmt"

// Inherits a Person object, and adds on a staffNumber
type Employee struct {
	Person
	staffNumber uint
}

// Returns a string representation of the employee
func (employee Employee) String() string {
	return fmt.Sprintf("%s, %d", employee.Person, employee.staffNumber)
}
